src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -37,16 +39,34 @@ function createExternalTextureSamplingTestPipeline(
     vertex: {
       module: t.device.createShaderModule({
         code: `
-        @vertex fn main(@builtin(vertex_index) VertexIndex : u32) -> @builtin(position) vec4<f32> {
-            var pos = array<vec4<f32>, 6>(
-              vec4<f32>( 1.0,  1.0, 0.0, 1.0),
-              vec4<f32>( 1.0, -1.0, 0.0, 1.0),
-              vec4<f32>(-1.0, -1.0, 0.0, 1.0),
-              vec4<f32>( 1.0,  1.0, 0.0, 1.0),
-              vec4<f32>(-1.0, -1.0, 0.0, 1.0),
-              vec4<f32>(-1.0,  1.0, 0.0, 1.0)
+        struct VertexOutput {
+          @builtin(position) Position : vec4f,
+          @location(0) fragUV : vec2f,
+        }
+
+        @vertex fn main(@builtin(vertex_index) VertexIndex : u32) -> VertexOutput {
+            const pos = array(
+              vec2( 1.0,  1.0),
+              vec2( 1.0, -1.0),
+              vec2(-1.0, -1.0),
+              vec2( 1.0,  1.0),
+              vec2(-1.0, -1.0),
+              vec2(-1.0,  1.0),
+            );
+
+            const uv = array(
```

Comment:
note that the uv could be generate from `(pos + vec2(1)) / 2`

Reply:
But also why did this need to change compared to the previous approach used fragcoord?

Reply:
In previous shader code, frag use fragcoord and calculate uv coord by setting texture size. Instead of using uniform to pass the texture size, I think maybe this way could do better.